expipe/core.py

```diff
@@ -1175,144 +765,109 @@ def dtype_manager(self, value, iter_value=False, retrieve=False):
-            result = new_result
-        except AttributeError:
-            pass
+            return self.create_project(name)
+
+
+    def delete_project(self, name, remove_all_children=None):
+        if self._backend.exists(name):
+            self._backend.delete_project(name, remove_all_children=remove_all_children)
+        else:
+            raise KeyError("Project does not exist.")
+
+# Entry API
+def get_project(path, name=None):
+    import expipe.backends.filesystem
+    path = pathlib.Path(path)
+
+    name = name or path.stem
+
+    global_config = config.settings.copy()
+
+    local_config_path = path / "expipe.yaml"
+    if not local_config_path.exists():
+        raise KeyError("Could not find '{}'.".format(local_config_path))
+
+    local_config = config._load_config(local_config_path)
+
     try:
-        if not isinstance(value, list) and np.isnan(result):
-            result = 'NaN'
-    except TypeError:
-        pass
-    return result
-
-
-def get_project(project_id):
-    project_db = FirebaseBackend("/projects")
-    existing = project_db.exists(project_id)
-    if not existing:
-        raise KeyError("Project " + project_id + " does not exist.")
-    return Project(project_id)
-
-
-def require_project(project_id):
-    """Creates a new project with the provided id."""
-    project_db = FirebaseBackend("/projects")
-    existing = project_db.exists(project_id)
-    registered = dt.datetime.today().strftime(datetime_format)
-    if not existing:
-        project_db.set(name=project_id, value={"registered": registered})
-
-    return Project(project_id)
-
-
-def delete_project(project_id, remove_all_childs=False):
-    """Deletes a project named after the provided id."""
-    project_db = FirebaseBackend("/projects")
-    existing = project_db.exists(project_id)
-    if not existing:
-        raise NameError('Project "' + project_id + '" does not exist.')
-    else:
-        if remove_all_childs:
-            project = Project(project_id)
-            for action in list(project.actions.keys()):
-                project.delete_action(action)
-            for module in list(project.modules.keys()):
-                project.delete_module(module)
-            for entity in list(project.entities.keys()):
-                project.delete_entity(entity)
-            for template in list(project.templates.keys()):
-                project.delete_template(template)
-        project_db.delete(name=project_id)
+        project = local_config['project']
+        project_config = config._load_config_by_name(project)
+    except KeyError:
+        warnings.warn("Project has no name. Please add 'project: name' to '{}'.".format(local_config_path))
+        project = "unnamed"
+        project_config = {}
+
+    if name is not None and name != project:
+        warnings.warn("Requested project with name '{}', but found '{}'".format(name, project))
+
+    final_config = {**global_config, **project_config, **local_config}
```

Comment:
Ah, yes, that probably fails pretty badly now. Is this something that Python has built-in somewhere? To update a dict by appending instead of overwriting whenever a list value is encountered?

Reply:
fixed